src/ipred.rs

```diff
@@ -1198,87 +1198,82 @@ unsafe fn filter_fn(
     p6: c_int,
 ) -> c_int {
     if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
```

Comment:
To elide a bunch of bounds checks.

```suggestion
    let flt_ptr = &flt_ptr[..48 + 1];
    if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
```